src/tests/functional/shared_test_classes/src/single_layer/eye.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "shared_test_classes/single_layer/eye.hpp"
+
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include <ngraph/opsets/opset9.hpp>
+#include <openvino/op/parameter.hpp>
+#include <openvino/pass/constant_folding.hpp>
+
+#include "ngraph_functions/builders.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string EyeLayerGPUTest::getTestCaseName(testing::TestParamInfo<EyeLayerTestParams> obj) {
+    EyeLayerTestParams params = obj.param;
+
+    std::string td;
+    std::vector<ov::Shape> input_shapes;
+    ElementType net_precision;
+    std::vector<LocalElementType> out_batch_shape;
+    std::vector<int> eye_par;
+    std::tie(input_shapes, out_batch_shape, eye_par, net_precision, td) = params;
+    std::ostringstream result;
+    result << "EyeTest_";
+    result << "IS=(";
+    for (const auto& shape : input_shapes) {
+        result << CommonTestUtils::partialShape2str({shape}) << "_";
+    }
+    result << ")";
+    result << "rowNum=" << eye_par[0] << "_";
+    result << "colNum=" << eye_par[1] << "_";
+    result << "diagShift=" << eye_par[2] << "_";
+    result << "batchShape=" << CommonTestUtils::vec2str(out_batch_shape) << "_";
+    result << net_precision << "_";
+    result << std::to_string(obj.index);
+    return result.str();
+}
+
+void EyeLayerGPUTest::SetUp() {
```

Comment:
I think GPU specific test class should be located in plugin/gpu/single_layer_test. There is no shared test class of eye_like yet.